### ISSUE TITLE
Styles appear in the legend of the example site, but not locally

### ISSUE THREAD

**Issue by YoeriVD** [avisi-cloud/structurizr-site-generatr#163]
The example website displays a beautiful feature of generating a legend, I assume based on the styles.
![image](https://user-images.githubusercontent.com/2865793/222735038-36d01882-a239-4f70-a382-a3e41fca656a.png)

However this does not seem to work in my case:

![image](https://user-images.githubusercontent.com/2865793/222735487-4b76d721-22ce-4236-a5f1-7ba5944bb37a.png)

```
        mySystem = softwareSystem "My System" "some system" "" {
            cont = container "My Container" "Some container" "" "existing_system"
            db = container "Database" "Stores all the things" "" "Database"
        }
    views {
      styles {
          relationship "existing_system" {
              color #EE0000
          }
      }
   }
```

It might be an error or something I missed, but unfortunately I have no idea which 🤷‍♂️.

Any pointers would be much appreciated!

I have updated a previous repro example to reflect this:
https://github.com/YoeriVD/structurizr-site-generatr-reproduction-repo

**Comment by jp7677** [comment]
Yeah, the styling is sometimes a bit of a mystery. If you want to style based on custom tag names, you'll need to enable tags for plantuml exporter, thus

```
properties {
    "c4plantuml.tags" "true"
}
``` 

(placed inside `views`) should help.

**Comment by qtzar** [comment]
I must be doing something completely wrong because I can't get styling to work at all. Even a simple style to set the shape of a container doesn't seem to do anything.

```
OIDS = container "OI Datastore" "SQL" "Database"
```

```
styles {
  element "Database" {
  shape Cylinder
}
```

**Comment by YoeriVD** [comment]
> I must be doing something completely wrong because I can't get styling to work at all. Even a simple style to set the shape of a container doesn't seem to do anything.
> 
> 
> 
> ```
> 
> OIDS = container "OI Datastore" "SQL" "Database"
> 
> ```
> 
> 
> 
> ```
> 
> styles {
> 
>   element "Database" {
> 
>   shape Cylinder
> 
> }
> 
> ```

Try this:

```
OIDS = container "OI Datastore" "my description" "SQL" "Database"
```

Tags define styling afaik and tags are the 4th parameter of a container.

**Comment by jp7677** [comment]
> The custom theme (azure icons), however, is not loading. Is that still an error or something that isn't supported by the site generator?

I’m not really familiar with the theming, but I think this is a limitation of the C4-Plantuml exporter (https://github.com/structurizr/export) that this project uses. Basic theming is supposed to work, see this post https://github.com/avisi-cloud/structurizr-site-generatr/issues/40#issuecomment-1386692888 but when trying your example here https://structurizr.com/dsl the icons also don’t appear in the rendered C4-Plantutml diagram.

**Comment by simonbrowndotje** [comment]
> I’m not really familiar with the theming, but I think this is a limitation of the C4-Plantuml exporter

Correct, the [C4PlantUMLExporter](https://github.com/structurizr/export/tree/main/src/main/java/com/structurizr/export/plantuml#c4plantumlexporter) doesn't support image icons. If there's a way to do this, please let me know and I'll add it. Icons are implemented in the [StructurizrPlantUMLExporter](https://github.com/structurizr/export/tree/main/src/main/java/com/structurizr/export/plantuml#structurizrplantumlexporter) though:

<img width="1728" alt="Screenshot 2023-03-10 at 06 46 07" src="https://user-images.githubusercontent.com/1009874/224243780-7bfda9c7-ca9f-4b17-b314-d67b7326140b.png">

As it says at [https://github.com/structurizr/export#diagram-notation](https://github.com/structurizr/export#diagram-notation) ... the PlantUML exporters try to replicate the Structurizr browser-based renderer, but many features are not possible with a format like PlantUML. For the richest diagrams, you should use the Structurizr renderer, which itself is now open source -> https://github.com/structurizr/ui

**Comment by jp7677** [comment]
Thanks a lot for the clarification!

(PS: Sorry if my statement came over as "something is _wrong_ with your exporter library", what I meant was: "this is probably outside the scope of this project with the currently chosen exporter".)

**Comment by simonbrowndotje** [comment]
> Yeah, the styling is sometimes a bit of a mystery.

I have added a table to https://github.com/structurizr/export#diagram-notation that summarises the feature differences between the exporters.

Styling is also documented in several places:

- https://structurizr.com/help/notation
- https://github.com/structurizr/dsl/blob/master/docs/language-reference.md#styles
- https://github.com/structurizr/dsl/tree/master/docs/cookbook/element-styles

**Comment by simonbrowndotje** [comment]
> "this is probably outside the scope of this project with the currently chosen exporter"

You may want to add some documentation to the repo README telling users that Site Generatr uses the C4PlantUMLExporter, and to not expect the full styling feature set provided by the Structurizr DSL.

**Comment by simonbrowndotje** [comment]
`structurizr-export:1.11.0` has been released; demo -> https://structurizr.com/dsl?src=https://raw.githubusercontent.com/YoeriVD/structurizr-site-generatr-reproduction-repo/main/workspace.dsl&renderer=c4plantuml&view=ProductionDeployment
